plugins/audio-s5gen2/fu-audio-s5gen2-hid-device.c

```diff
@@ -0,0 +1,291 @@
+/*
+ * Copyright (C) 2023 Denis Pynkin <denis.pynkin@collabora.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include "fu-audio-s5gen2-device.h"
+#include "fu-audio-s5gen2-hid-device.h"
+#include "fu-audio-s5gen2-hid-struct.h"
+#include "fu-audio-s5gen2-struct.h"
+
+#define HID_IFACE  0x01
+#define HID_EP_IN  0x82
+#define HID_EP_OUT 0x01
+
+/* FIXME: values :-| */
+#define FU_QC_S5GEN2_HID_DEVICE_TIMEOUT	     0	   /* ms */
+#define FU_QC_S5GEN2_HID_DEVICE_REMOVE_DELAY 60000 /* ms */
+
+struct _FuQcS5gen2HidDevice {
+	FuHidDevice parent_instance;
+	FuDevice *core;
```

Comment:
I think rather than storing this here you want to call `fu_device_set_proxy()` when it gets created. Then it gets lifecycled with the main object too.